odxtools/parameters/tablekeyparameter.py

```diff
@@ -37,37 +40,107 @@ def _resolve_odxlinks(self, odxlinks: OdxLinkDatabase) -> None:
         super()._resolve_odxlinks(odxlinks)
 
         if self.table_ref:
-            self._table = odxlinks.resolve(self.table_ref)
+            if TYPE_CHECKING:
+                self._table = odxlinks.resolve(self.table_ref, Table)
+            else:
+                self._table = odxlinks.resolve(self.table_ref)
+
+        self._table_row: Optional[TableRow] = None
         if self.table_row_ref:
-            self._table_row = odxlinks.resolve(self.table_row_ref)
+            if TYPE_CHECKING:
+                self._table_row = odxlinks.resolve(self.table_row_ref, TableRow)
+            else:
+                self._table_row = odxlinks.resolve(self.table_row_ref)
```

Comment:
Why different code based on TYPE_CHECKING ?

Reply:
to prevent mypy from complaining without getting circular imports.